### DISCUSSION TITLE
Curious about Quarkus' choice of default package structure

### DISCUSSION THREAD

**Discussion by preslavrachev** [quarkusio/quarkus#39910] [Q&A]
Below are two identical barebones starting projects in Quarkus and Spring Boot, respectively. Notice, how Spring follows the (to my understanding) more common 3-level nested approach (domain, organization, project), while Quarkus omits the project level entirely.

I am trying to address those differences to a Spring developer, thus I'd like to know, was that an intentional decision, or some sort of internal RedHat convention that the Quarkus team just stuck with?

![SCR-20240405-ossh](https://github.com/quarkusio/quarkus/assets/86060/522591f4-c43a-4cf9-b68e-d1088c1f72b2)

**Comment by gsmet** [comment]
I think it's more that you will get rid of this package anyway as you will want your own package name.
So not sure it's worth making them more complicated than they need.

Now maybe we could add the project name here. Not entirely sure how easy it would be.

**Reply by preslavrachev** [reply]
NOTE: I updated the screenshot, because it wasn't clear which package I mean. So, while in Spring you'd have `com.example.demo`, In its Quarkus equivalent, that would be `com.example`.

At this point, I just wantted to clarify this difference, as I am used to working with projects following the 3-level nesting. It's more of a habit than a necessity, really, but I am frequently being asked by people working with Spring so I was obliged to forward the question to the Quarkus team ;)
